docs/extensions/kuneiform-precompiles/precompiles.mdx

```diff
@@ -0,0 +1,142 @@
+---
+sidebar_position: 1
+sidebar_label: "Precompiles"
+id: precompiles
+title: Kuneiform Precompiles
+description: Extend Kuneiform with custom precompiles.
+slug: /extensions/precompiles
+---
+
+Kuneiform precompiles are custom libraries that can be imported and called in Kuneiform actions.
+By implementing custom action precompiles, developers can add support for new functionality to their Kwil network's Kuneiform language.
+These functionalities are wide ranging, and can include anything from performing hashing algorithms to directly calling the active Postgres connection.
+
+## How Precompiles Work
+
+Action precompiles are implemented as Go libraries that are loaded into the Kwil node at startup. All precompiles are globally available to all Kuneiform schemas.
+There are two distinct steps in an action precompile's lifecycle:
+
+1. **Initialize**: The action precompile is imported into a Kuneiform schema. It is given a set of initialization values, and a name by which it can be called. Initialize is called
+_each time_ the precompile is imported into a schema. If a node is restarted, all precompiles for all persisted schemas are re-initialized.
+
+2. **Call**: The action precompile is called by its name (given during initialization), and is given a method name and a set of arguments.
+It then performs the requested action and returns a result. This can only be done after the precompile has been initialized.
```

Comment:
```suggestion
It then performs the requested action and returns a result to the Kuneiform schema. This can only be done after the precompile has been initialized.
```

Reply:
`Kuneiform Schema` shouldn't refer to deployed databases. A schema is simply that- the schema for a database